In [7]:
import pandas as pd
star_wars = pd.read_csv("star_wars_character_dataset.csv")
star_wars

name  height   mass hair_color   skin_color eye_color  \
0   Luke Skywalker   172.0   77.0      blond         fair      blue   
1            C-3PO   167.0   75.0        NaN         gold    yellow   
2            R2-D2    96.0   32.0        NaN  white, blue       red   
3      Darth Vader   202.0  136.0       none        white    yellow   
4      Leia Organa   150.0   49.0      brown        light     brown   
..             ...     ...    ...        ...          ...       ...   
82             Rey     NaN    NaN      brown        light     hazel   
83     Poe Dameron     NaN    NaN      brown        light     brown   
84             BB8     NaN    NaN       none         none     black   
85  Captain Phasma     NaN    NaN    unknown      unknown   unknown   
86   Padmé Amidala   165.0   45.0      brown        light     brown   

    birth_year     sex     gender homeworld species  \
0         19.0    male  masculine  Tatooine   Human   
1        112.0    none  masculine  Tatooine   Droid   
2         33.0    none  masculine     Naboo   Droid   
3         41.9    male  masculine  Tatooine   Human   
4         19.0  female   feminine  Alderaan   Human   
..         ...     ...        ...       ...     ...   
82         NaN  female   feminine       NaN   Human   
83         NaN    male  masculine       NaN   Human   
84         NaN    none  masculine       NaN   Droid   
85         NaN     NaN        NaN       NaN     NaN   
86        46.0  female   feminine     Naboo   Human   

                                                films  \
0   The Empire Strikes Back, Revenge of the Sith, ...   
1   The Empire Strikes Back, Attack of the Clones,...   
2   The Empire Strikes Back, Attack of the Clones,...   
3   The Empire Strikes Back, Revenge of the Sith, ...   
4   The Empire Strikes Back, Revenge of the Sith, ...   
..                                                ...   
82                                  The Force Awakens   
83                                  The Force Awakens   
84                                  The Force Awakens   
85                                  The Force Awakens   
86  Attack of the Clones, The Phantom Menace, Reve...   

                              vehicles  \
0   Snowspeeder, Imperial Speeder Bike   
1                                  NaN   
2                                  NaN   
3                                  NaN   
4                Imperial Speeder Bike   
..                                 ...   
82                                 NaN   
83                                 NaN   
84                                 NaN   
85                                 NaN   
86                                 NaN   

                                            starships  
0                            X-wing, Imperial shuttle  
1                                                 NaN  
2                                                 NaN  
3                                     TIE Advanced x1  
4                                                 NaN  
..                                                ...  
82                                                NaN  
83                                T-70 X-wing fighter  
84                                                NaN  
85                                                NaN  
86  H-type Nubian yacht, Naboo star skiff, Naboo f...  

[87 rows x 14 columns]

1. Вывод данных.
Выведем полный датасет со всеми данными. Всего 87 персонажей. Более подробный разбор
датасета будет в пункте про наименование колонок.

In [8]:
star_wars.shape

(87, 14)

2. Размерность массива
Размерность показывает количество строк и столбцов таблицы. В нашем случае общее количество ячеек будет 1218, столько понадобится индексов, чтобы определить каждый элемент массива.

In [9]:
  star_wars.columns

Index(['name', 'height', 'mass', 'hair_color', 'skin_color', 'eye_color',
       'birth_year', 'sex', 'gender', 'homeworld', 'species', 'films',
       'vehicles', 'starships'],
      dtype='object')

3. Наименования колонок
В нашем датасете указаны такие данные персонажей, как имя, рост, вес, цвет волос, цыет
кожи, цвет глаз, день рождеия, пол, гендер, родная планета, разновидность, фильмы
и транспорт с космическими кораблями по которым мы будем сравнивать персонажей.

In [10]:
star_wars['name'].unique()

array(['Luke Skywalker', 'C-3PO', 'R2-D2', 'Darth Vader', 'Leia Organa',
       'Owen Lars', 'Beru Whitesun lars', 'R5-D4', 'Biggs Darklighter',
       'Obi-Wan Kenobi', 'Anakin Skywalker', 'Wilhuff Tarkin',
       'Chewbacca', 'Han Solo', 'Greedo', 'Jabba Desilijic Tiure',
       'Wedge Antilles', 'Jek Tono Porkins', 'Yoda', 'Palpatine',
       'Boba Fett', 'IG-88', 'Bossk', 'Lando Calrissian', 'Lobot',
       'Ackbar', 'Mon Mothma', 'Arvel Crynyd', 'Wicket Systri Warrick',
       'Nien Nunb', 'Qui-Gon Jinn', 'Nute Gunray', 'Finis Valorum',
       'Jar Jar Binks', 'Roos Tarpals', 'Rugor Nass', 'Ric Olié', 'Watto',
       'Sebulba', 'Quarsh Panaka', 'Shmi Skywalker', 'Darth Maul',
       'Bib Fortuna', 'Ayla Secura', 'Dud Bolt', 'Gasgano',
       'Ben Quadinaros', 'Mace Windu', 'Ki-Adi-Mundi', 'Kit Fisto',
       'Eeth Koth', 'Adi Gallia', 'Saesee Tiin', 'Yarael Poof',
       'Plo Koon', 'Mas Amedda', 'Gregar Typho', 'Cordé', 'Cliegg Lars',
       'Poggle the Lesser', 'Luminara Unduli'

In [11]:
star_wars['films'].unique()

array(['The Empire Strikes Back, Revenge of the Sith, Return of the Jedi, A New Hope, The Force Awakens',
       'The Empire Strikes Back, Attack of the Clones, The Phantom Menace, Revenge of the Sith, Return of the Jedi, A New Hope',
       'The Empire Strikes Back, Attack of the Clones, The Phantom Menace, Revenge of the Sith, Return of the Jedi, A New Hope, The Force Awakens',
       'The Empire Strikes Back, Revenge of the Sith, Return of the Jedi, A New Hope',
       'Attack of the Clones, Revenge of the Sith, A New Hope',
       'A New Hope',
       'Attack of the Clones, The Phantom Menace, Revenge of the Sith',
       'Revenge of the Sith, A New Hope',
       'The Empire Strikes Back, Return of the Jedi, A New Hope, The Force Awakens',
       'The Phantom Menace, Return of the Jedi, A New Hope',
       'The Empire Strikes Back, Return of the Jedi, A New Hope',
       'The Empire Strikes Back, Attack of the Clones, The Phantom Menace, Revenge of the Sith, Return of the Jedi',
  

In [12]:
star_wars['starships'].unique()

array(['X-wing, Imperial shuttle', nan, 'TIE Advanced x1', 'X-wing',
       'Jedi starfighter, Trade Federation cruiser, Naboo star skiff, Jedi Interceptor, Belbullab-22 starfighter',
       'Trade Federation cruiser, Jedi Interceptor, Naboo fighter',
       'Millennium Falcon, Imperial shuttle', 'Slave 1',
       'Millennium Falcon', 'A-wing', 'Naboo Royal Starship', 'Scimitar',
       'Jedi starfighter', 'Naboo fighter', 'Belbullab-22 starfighter',
       'T-70 X-wing fighter',
       'H-type Nubian yacht, Naboo star skiff, Naboo fighter'],
      dtype=object)

In [13]:
star_wars['birth_year'].unique()

array([ 19. , 112. ,  33. ,  41.9,  52. ,  47. ,   nan,  24. ,  57. ,
        64. , 200. ,  29. ,  44. , 600. ,  21. , 896. ,  82. ,  31.5,
        15. ,  53. ,  31. ,  37. ,  41. ,  48. ,   8. ,  92. ,  91. ,
        62. ,  72. ,  54. ,  22. ,  58. ,  40. , 102. ,  67. ,  66. ,
        46. ])

In [14]:
star_wars['hair_color'].unique()

array(['blond', nan, 'none', 'brown', 'brown, grey', 'black',
       'auburn, white', 'auburn, grey', 'white', 'grey', 'auburn',
       'blonde', 'unknown'], dtype=object)

In [15]:
star_wars['skin_color'].unique()

array(['fair', 'gold', 'white, blue', 'white', 'light', 'white, red',
       'unknown', 'green', 'green-tan, brown', 'pale', 'metal', 'dark',
       'brown mottle', 'brown', 'grey', 'mottled green', 'orange',
       'blue, grey', 'grey, red', 'red', 'blue', 'grey, green, yellow',
       'yellow', 'tan', 'fair, green, yellow', 'grey, blue',
       'silver, red', 'green, grey', 'red, blue, white', 'brown, white',
       'none'], dtype=object)

In [18]:
star_wars['mass'].unique()

array([  77. ,   75. ,   32. ,  136. ,   49. ,  120. ,   84. ,    nan,
        112. ,   80. ,   74. , 1358. ,  110. ,   17. ,   78.2,  140. ,
        113. ,   79. ,   83. ,   20. ,   68. ,   89. ,   90. ,   66. ,
         82. ,   40. ,   55. ,   45. ,   65. ,   87. ,   50. ,   85. ,
         56.2,  102. ,   88. ,   15. ,   48. ,   57. ,  159. ])

In [17]:
star_wars['species'].unique()

array(['Human', 'Droid', 'Wookiee', 'Rodian', 'Hutt', "Yoda's species",
       'Trandoshan', 'Mon Calamari', 'Ewok', 'Sullustan', 'Neimodian',
       'Gungan', nan, 'Toydarian', 'Dug', 'Zabrak', "Twi'lek",
       'Vulptereen', 'Xexto', 'Toong', 'Cerean', 'Nautolan', 'Tholothian',
       'Iktotchi', 'Quermian', 'Kel Dor', 'Chagrian', 'Geonosian',
       'Mirialan', 'Clawdite', 'Besalisk', 'Kaminoan', 'Aleena',
       'Skakoan', 'Muun', 'Togruta', 'Kaleesh', "Pau'an"], dtype=object)

In [19]:
star_wars['homeworld'].unique()

array(['Tatooine', 'Naboo', 'Alderaan', 'Stewjon', 'Eriadu', 'Kashyyyk',
       'Corellia', 'Rodia', 'Nal Hutta', 'Bestine IV', nan, 'Kamino',
       'Trandosha', 'Socorro', 'Bespin', 'Mon Cala', 'Chandrila', 'Endor',
       'Sullust', 'Cato Neimoidia', 'Coruscant', 'Toydaria', 'Malastare',
       'Dathomir', 'Ryloth', 'Vulpter', 'Troiken', 'Tund', 'Haruun Kal',
       'Cerea', 'Glee Anselm', 'Iridonia', 'Iktotch', 'Quermia', 'Dorin',
       'Champala', 'Geonosis', 'Mirial', 'Serenno', 'Concord Dawn',
       'Zolan', 'Ojom', 'Aleen Minor', 'Skako', 'Muunilinst', 'Shili',
       'Kalee', 'Umbara', 'Utapau'], dtype=object)

In [20]:
star_wars['vehicles'].unique()

array(['Snowspeeder, Imperial Speeder Bike', nan, 'Imperial Speeder Bike',
       'Tribubble bongo', 'Zephyr-G swoop bike, XJ-6 airspeeder', 'AT-ST',
       'Snowspeeder', 'Sith speeder', 'Flitknot speeder',
       'Koro-2 Exodrive airspeeder', 'Tsmeu-6 personal wheel bike'],
      dtype=object)

In [23]:
star_wars['sex'].unique()

array(['male', 'none', 'female', 'hermaphroditic', nan], dtype=object)

In [24]:
star_wars.nunique()

name          87
height        45
mass          38
hair_color    12
skin_color    31
eye_color     15
birth_year    36
sex            4
gender         2
homeworld     48
species       37
films         24
vehicles      10
starships     16
dtype: int64

4. Уникальные значения
Выводим уникальные значения для каждой колонки, а также общее количество уникальных
значений в датасете.

In [25]:
star_wars.sort_values(by='films')

name  height    mass hair_color        skin_color  \
17       Jek Tono Porkins   180.0   110.0      brown              fair   
14                 Greedo   173.0    74.0        NaN             green   
7                   R5-D4    97.0    32.0        NaN        white, red   
8       Biggs Darklighter   183.0    84.0      black             light   
58            Cliegg Lars   183.0     NaN      brown              fair   
..                    ...     ...     ...        ...               ...   
36               Ric Olié   183.0     NaN      brown              fair   
15  Jabba Desilijic Tiure   175.0  1358.0        NaN  green-tan, brown   
50              Eeth Koth   171.0     NaN      black             brown   
51             Adi Gallia   184.0    50.0       none              dark   
52            Saesee Tiin   188.0     NaN       none              pale   

   eye_color  birth_year             sex     gender   homeworld     species  \
17      blue         NaN            male  masculine  Bestine IV       Human   
14     black        44.0            male  masculine       Rodia      Rodian   
7        red         NaN            none  masculine    Tatooine       Droid   
8      brown        24.0            male  masculine    Tatooine       Human   
58      blue        82.0            male  masculine    Tatooine       Human   
..       ...         ...             ...        ...         ...         ...   
36      blue         NaN             NaN        NaN       Naboo         NaN   
15    orange       600.0  hermaphroditic  masculine   Nal Hutta        Hutt   
50     brown         NaN            male  masculine    Iridonia      Zabrak   
51      blue         NaN          female   feminine   Coruscant  Tholothian   
52    orange         NaN            male  masculine     Iktotch    Iktotchi   

                                                films vehicles  \
17                                         A New Hope      NaN   
14                                         A New Hope      NaN   
7                                          A New Hope      NaN   
8                                          A New Hope      NaN   
58                               Attack of the Clones      NaN   
..                                                ...      ...   
36                                 The Phantom Menace      NaN   
15  The Phantom Menace, Return of the Jedi, A New ...      NaN   
50            The Phantom Menace, Revenge of the Sith      NaN   
51            The Phantom Menace, Revenge of the Sith      NaN   
52            The Phantom Menace, Revenge of the Sith      NaN   

               starships  
17                X-wing  
14                   NaN  
7                    NaN  
8                 X-wing  
58                   NaN  
..                   ...  
36  Naboo Royal Starship  
15                   NaN  
50                   NaN  
51                   NaN  
52                   NaN  

[87 rows x 14 columns]

In [26]:
star_wars.sort_values(by='birth_year')

name  height   mass hair_color skin_color eye_color  \
28  Wicket Systri Warrick    88.0   20.0      brown      brown     brown   
21                  IG-88   200.0  140.0       none      metal       red   
0          Luke Skywalker   172.0   77.0      blond       fair      blue   
4             Leia Organa   150.0   49.0      brown      light     brown   
16         Wedge Antilles   170.0   77.0      brown       fair     hazel   
..                    ...     ...    ...        ...        ...       ...   
81                   Finn     NaN    NaN      black       dark      dark   
82                    Rey     NaN    NaN      brown      light     hazel   
83            Poe Dameron     NaN    NaN      brown      light     brown   
84                    BB8     NaN    NaN       none       none     black   
85         Captain Phasma     NaN    NaN    unknown    unknown   unknown   

    birth_year     sex     gender homeworld species  \
28         8.0    male  masculine     Endor    Ewok   
21        15.0    none  masculine       NaN   Droid   
0         19.0    male  masculine  Tatooine   Human   
4         19.0  female   feminine  Alderaan   Human   
16        21.0    male  masculine  Corellia   Human   
..         ...     ...        ...       ...     ...   
81         NaN    male  masculine       NaN   Human   
82         NaN  female   feminine       NaN   Human   
83         NaN    male  masculine       NaN   Human   
84         NaN    none  masculine       NaN   Droid   
85         NaN     NaN        NaN       NaN     NaN   

                                                films  \
28                                 Return of the Jedi   
21                            The Empire Strikes Back   
0   The Empire Strikes Back, Revenge of the Sith, ...   
4   The Empire Strikes Back, Revenge of the Sith, ...   
16  The Empire Strikes Back, Return of the Jedi, A...   
..                                                ...   
81                                  The Force Awakens   
82                                  The Force Awakens   
83                                  The Force Awakens   
84                                  The Force Awakens   
85                                  The Force Awakens   

                              vehicles                 starships  
28                                 NaN                       NaN  
21                                 NaN                       NaN  
0   Snowspeeder, Imperial Speeder Bike  X-wing, Imperial shuttle  
4                Imperial Speeder Bike                       NaN  
16                         Snowspeeder                    X-wing  
..                                 ...                       ...  
81                                 NaN                       NaN  
82                                 NaN                       NaN  
83                                 NaN       T-70 X-wing fighter  
84                                 NaN                       NaN  
85                                 NaN                       NaN  

[87 rows x 14 columns]

5. Сортировка.
Отсортируем наш датасет по фильмам и году рождения 

In [28]:
star_wars.drop(star_wars.columns[[5,6,7,9,10,11,12,13]], axis=1)

name  height   mass hair_color   skin_color     gender
0   Luke Skywalker   172.0   77.0      blond         fair  masculine
1            C-3PO   167.0   75.0        NaN         gold  masculine
2            R2-D2    96.0   32.0        NaN  white, blue  masculine
3      Darth Vader   202.0  136.0       none        white  masculine
4      Leia Organa   150.0   49.0      brown        light   feminine
..             ...     ...    ...        ...          ...        ...
82             Rey     NaN    NaN      brown        light   feminine
83     Poe Dameron     NaN    NaN      brown        light  masculine
84             BB8     NaN    NaN       none         none  masculine
85  Captain Phasma     NaN    NaN    unknown      unknown        NaN
86   Padmé Amidala   165.0   45.0      brown        light   feminine

[87 rows x 6 columns]

6. Удаление
Удалим ненужные стобцы, чтобы проще было ориентироваться.

In [29]:
star_wars = star_wars.replace({"vehicles":"NaN"}, "Not Found")
star_wars

name  height   mass hair_color   skin_color eye_color  \
0   Luke Skywalker   172.0   77.0      blond         fair      blue   
1            C-3PO   167.0   75.0        NaN         gold    yellow   
2            R2-D2    96.0   32.0        NaN  white, blue       red   
3      Darth Vader   202.0  136.0       none        white    yellow   
4      Leia Organa   150.0   49.0      brown        light     brown   
..             ...     ...    ...        ...          ...       ...   
82             Rey     NaN    NaN      brown        light     hazel   
83     Poe Dameron     NaN    NaN      brown        light     brown   
84             BB8     NaN    NaN       none         none     black   
85  Captain Phasma     NaN    NaN    unknown      unknown   unknown   
86   Padmé Amidala   165.0   45.0      brown        light     brown   

    birth_year     sex     gender homeworld species  \
0         19.0    male  masculine  Tatooine   Human   
1        112.0    none  masculine  Tatooine   Droid   
2         33.0    none  masculine     Naboo   Droid   
3         41.9    male  masculine  Tatooine   Human   
4         19.0  female   feminine  Alderaan   Human   
..         ...     ...        ...       ...     ...   
82         NaN  female   feminine       NaN   Human   
83         NaN    male  masculine       NaN   Human   
84         NaN    none  masculine       NaN   Droid   
85         NaN     NaN        NaN       NaN     NaN   
86        46.0  female   feminine     Naboo   Human   

                                                films  \
0   The Empire Strikes Back, Revenge of the Sith, ...   
1   The Empire Strikes Back, Attack of the Clones,...   
2   The Empire Strikes Back, Attack of the Clones,...   
3   The Empire Strikes Back, Revenge of the Sith, ...   
4   The Empire Strikes Back, Revenge of the Sith, ...   
..                                                ...   
82                                  The Force Awakens   
83                                  The Force Awakens   
84                                  The Force Awakens   
85                                  The Force Awakens   
86  Attack of the Clones, The Phantom Menace, Reve...   

                              vehicles  \
0   Snowspeeder, Imperial Speeder Bike   
1                                  NaN   
2                                  NaN   
3                                  NaN   
4                Imperial Speeder Bike   
..                                 ...   
82                                 NaN   
83                                 NaN   
84                                 NaN   
85                                 NaN   
86                                 NaN   

                                            starships  
0                            X-wing, Imperial shuttle  
1                                                 NaN  
2                                                 NaN  
3                                     TIE Advanced x1  
4                                                 NaN  
..                                                ...  
82                                                NaN  
83                                T-70 X-wing fighter  
84                                                NaN  
85                                                NaN  
86  H-type Nubian yacht, Naboo star skiff, Naboo f...  

[87 rows x 14 columns]

7. Замена 
Заменим значения NaN в колонке транспорта на Not Found.

In [30]:
star_wars.drop_duplicates()

name  height   mass hair_color   skin_color eye_color  \
0   Luke Skywalker   172.0   77.0      blond         fair      blue   
1            C-3PO   167.0   75.0        NaN         gold    yellow   
2            R2-D2    96.0   32.0        NaN  white, blue       red   
3      Darth Vader   202.0  136.0       none        white    yellow   
4      Leia Organa   150.0   49.0      brown        light     brown   
..             ...     ...    ...        ...          ...       ...   
82             Rey     NaN    NaN      brown        light     hazel   
83     Poe Dameron     NaN    NaN      brown        light     brown   
84             BB8     NaN    NaN       none         none     black   
85  Captain Phasma     NaN    NaN    unknown      unknown   unknown   
86   Padmé Amidala   165.0   45.0      brown        light     brown   

    birth_year     sex     gender homeworld species  \
0         19.0    male  masculine  Tatooine   Human   
1        112.0    none  masculine  Tatooine   Droid   
2         33.0    none  masculine     Naboo   Droid   
3         41.9    male  masculine  Tatooine   Human   
4         19.0  female   feminine  Alderaan   Human   
..         ...     ...        ...       ...     ...   
82         NaN  female   feminine       NaN   Human   
83         NaN    male  masculine       NaN   Human   
84         NaN    none  masculine       NaN   Droid   
85         NaN     NaN        NaN       NaN     NaN   
86        46.0  female   feminine     Naboo   Human   

                                                films  \
0   The Empire Strikes Back, Revenge of the Sith, ...   
1   The Empire Strikes Back, Attack of the Clones,...   
2   The Empire Strikes Back, Attack of the Clones,...   
3   The Empire Strikes Back, Revenge of the Sith, ...   
4   The Empire Strikes Back, Revenge of the Sith, ...   
..                                                ...   
82                                  The Force Awakens   
83                                  The Force Awakens   
84                                  The Force Awakens   
85                                  The Force Awakens   
86  Attack of the Clones, The Phantom Menace, Reve...   

                              vehicles  \
0   Snowspeeder, Imperial Speeder Bike   
1                                  NaN   
2                                  NaN   
3                                  NaN   
4                Imperial Speeder Bike   
..                                 ...   
82                                 NaN   
83                                 NaN   
84                                 NaN   
85                                 NaN   
86                                 NaN   

                                            starships  
0                            X-wing, Imperial shuttle  
1                                                 NaN  
2                                                 NaN  
3                                     TIE Advanced x1  
4                                                 NaN  
..                                                ...  
82                                                NaN  
83                                T-70 X-wing fighter  
84                                                NaN  
85                                                NaN  
86  H-type Nubian yacht, Naboo star skiff, Naboo f...  

[87 rows x 14 columns]

8. Удаление дубликатов 
Удалим дубликаты из датасета, если они есть. По количеству строк понимаем что не было.

In [31]:
star_wars.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 87 entries, 0 to 86
Data columns (total 14 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   name        87 non-null     object 
 1   height      81 non-null     float64
 2   mass        59 non-null     float64
 3   hair_color  82 non-null     object 
 4   skin_color  87 non-null     object 
 5   eye_color   87 non-null     object 
 6   birth_year  43 non-null     float64
 7   sex         83 non-null     object 
 8   gender      83 non-null     object 
 9   homeworld   77 non-null     object 
 10  species     83 non-null     object 
 11  films       87 non-null     object 
 12  vehicles    11 non-null     object 
 13  starships   20 non-null     object 
dtypes: float64(3), object(11)
memory usage: 9.6+ KB


9. Анализ с помощью info.
После info анализа можем сделать вывод что 20 персонажей имели свой космический корабль и  всего 11 имели свой транспорт  

In [32]:
star_wars.describe()

height         mass  birth_year
count   81.000000    59.000000   43.000000
mean   174.358025    97.311864   87.565116
std     34.770429   169.457163  154.691439
min     66.000000    15.000000    8.000000
25%    167.000000    55.600000   35.000000
50%    180.000000    79.000000   52.000000
75%    191.000000    84.500000   72.000000
max    264.000000  1358.000000  896.000000

10. Анализ с помощью Describe.
После describe анализа видим, что максимальный рост - 264, максимальный вес - 1358, минимальный год рождения - 8 лет, а максимальный - 896 лет.

In [33]:
star_wars.loc[[0,13,23],['vehicles', 'mass', 'sex', 'starships']]

vehicles  mass   sex  \
0   Snowspeeder, Imperial Speeder Bike  77.0  male   
13                                 NaN  80.0  male   
23                                 NaN  79.0  male   

                              starships  
0              X-wing, Imperial shuttle  
13  Millennium Falcon, Imperial shuttle  
23                    Millennium Falcon

11. Выборка с помощью loc.
Выберем 3 рандомных перса и просмотрим их транспорт, вес, пол и космический корабль.
Делаем вывод, что владелец снежного спидера и x-wing(а) более крутой чем челики без
своих ТС

In [35]:
star_wars.to_csv("starwars_new.csv", index=False)

12. Сохранение 
Сохраним обновленный датасет в новый csv

Вывод:был реализован датасет с персонажами Звездных Войн. Входе исследования оказалось
что не так много персонажей киносаги имеют свое личное ТС. Из представленных в 
датасете 87 только 11 имели свое ТС.Изучены функции describe, info, replace, drop,
drop_duplicates, loc, to_csv, sort_values, unique, shape, columns.